celest/units/quantity.py

```diff
@@ -0,0 +1,297 @@
+        if isinstance(other, (float, int)):
+            data = other / self._data
+            unit = 1 / self._unit
+            return Quantity(data, unit)
+        elif isinstance(other, Quantity):
+            if self._unit.dimension == other._unit.dimension:
+                data = other.to(self._unit) / self._data
+                unit = self._unit / self._unit
+            else:
+                data = other._data / self._data
+                unit = other._unit / self._unit
+            return Quantity(data, unit)
+        else:
+            return NotImplemented
+
+    def __eq__(self, other):
+        if isinstance(other, Quantity):
+            if self._unit.dimension != other._unit.dimension:
+                raise ArithmeticError(
+                    "Unit dimensions must match for comparison.")
+            return self._data == other.to(self._unit)
+        else:
+            return NotImplemented
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        if isinstance(other, Quantity):
+            if self._unit.dimension != other._unit.dimension:
+                raise ArithmeticError(
+                    "Unit dimensions must match for comparison.")
+            return self._data < other.to(self._unit)
+        else:
+            return NotImplemented
+
+    def __le__(self, other):
+        return self.__eq__(other) or self.__lt__(other)
+
+    def __gt__(self, other):
+        if isinstance(other, Quantity):
+            if self._unit.dimension != other._unit.dimension:
+                raise ArithmeticError(
+                    "Unit dimensions must match for comparison.")
+            return self._data > other.to(self._unit)
+        else:
+            return NotImplemented
+
+    def __ge__(self, other):
+        return self.__eq__(other) or self.__gt__(other)
+
+    @property
+    def data(self) -> Any:
+        """
+        Return the data of the quantity.
+        """
+        return self._data
+
+    @property
+    def unit(self) -> Union[Unit, CompoundUnit]:
+        """
+        Return the unit of the quantity.
+        """
+        return self._unit
+
+    @property
+    def dimension(self) -> str:
+        """
+        Return the dimension string of the quantity.
+        """
+        return self._unit.dimension
+
+    def to(self, new_unit: Union[Unit, CompoundUnit]) -> Any:
+        """Return the Quantity data in the new unit.
+
+        Parameters
+        ----------
+        new_unit : Unit, CompoundUnit
+            New unit with the same dimension as the current unit.
+
+        Returns
+        -------
+        Any
+            The Quantity data in the new unit.
+
+        Examples
+        --------
+        >>> quantity_in_meters = Quantity(5, u.m)
+        >>> quantity_in_meters.to(u.km)
+        0.005
+        """
+
+        return self._convert_to(new_unit).data
+
+    def _convert_to(self, new_unit: Union[Unit, CompoundUnit]) -> "Quantity":
```

Comment:
Make this public since you may want to change the internal dimension of a `Quantity` object for printing or saving to file.